packages/block-library/src/query-no-results/index.php

```diff
@@ -32,14 +32,10 @@ function render_block_core_query_no_results( $attributes, $content, $block ) {
 		$query      = new WP_Query( $query_args );
 	}
 
-	if ( $query->have_posts() ) {
+	if ( $query->post_count > 0 ) {
```

Comment:
```suggestion
	if ( $query->found_posts > 0 ) {
```

I'm wondering if it would be more appropriate to use `found_posts` instead of `post_count`, based on these descriptions from the [WP_Query docs](https://developer.wordpress.org/reference/classes/wp_query/#properties):

> `$post_count`
> The number of posts being displayed.
> `$found_posts`
> The total number of posts found matching the current query parameters

Reply:
Good question! `$post_count` is what `have_posts()` also uses internally. `$found_posts` includes the total count but in case of paginated content we have to use `$post_count` because we only care about the posts of the current page not all pages.

Reply:
Ah OK, thank you for explaining!